### PR TITLE
Modernized for not requiring URIJS anymore => reduces effect on bundle size

### PR DESCRIPTION
**Pull request by swernerx**
- Removes URIJS polyfill. This is natively supported by most browsers/environments today and reduces bundle size dramatically when using this library.
- Switches to `url-template` for templating needs.
- Test suite runs fine :) 
- Minor cleanup of `tsconfig`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,26 +24,22 @@
   ],
   "scripts": {
     "build": "tsc",
+    "check": "tsc --noEmit",
+    "clean": "git clean -fXd",
```
Comment (bastianeicher) [inline package.json]: Could you put this change into a separate commit, since it's not directory related to removing uri.js?

File: tsconfig.json [hunk 2]
```diff
@@ -7,13 +7,6 @@
     "noUnusedLocals": true,
     "strict": true
   },
-  "include": [
-    "**/*.ts"
-  ],
-  "exclude": [
-    "node_modules",
-    "**/*.test.ts",
-  ],
```
Comment (bastianeicher) [inline tsconfig.json]: Why is this change required? Wouldn't removing the exclusion potentially slow the build down a bit?

File: package.json [hunk 3]
```diff
@@ -24,26 +24,22 @@
...
-    "typedoc": "0.23.11",
-    "typescript": "4.8.3"
+    "@types/jest": "^29.0.2",
+    "@types/url-template": "^2.0.28",
+    "jest": "^29.0.3",
+    "jest-environment-jsdom": "^29.0.3",
+    "jest-fetch-mock": "^3.0.3",
+    "ts-jest": "^29.0.1",
+    "typedoc": "^0.23.14",
+    "typescript": "^4.8.3"
```
Comment (bastianeicher) [inline package.json]: I had previously pinned exact versions for dev dependencies, because Renovate keeps the up to date and this way builds are more reliably reproducible.

File: package.json [hunk 4]
```diff
@@ -24,26 +24,22 @@
   ],
   "scripts": {
     "build": "tsc",
+    "check": "tsc --noEmit",
+    "clean": "git clean -fXd",
     "test": "jest",
     "docs": "typedoc"
   },
-  "jest": {
-    "testEnvironment": "jsdom",
```
Comment (bastianeicher) [inline package.json:31]: Previously I needed to add this in order to make `fetch` work in the tests. Has something changed here?

File: package.json [hunk 5]
```diff
@@ -24,26 +24,22 @@
...
-    "transform": {
-      "\\.ts$": "ts-jest"
```
Comment (bastianeicher) [inline package.json:33]: Isn't this needed for Jest to fail on type errors?

**Review by bastianeicher — COMMENTED** [review]
Thank you for this great contribution!

I've just got a few small questions. Other than that, happy to merge this soon. :)

File: tsconfig.json (hunk 2 above)
Comment (swernerx) [reply tsconfig.json]: Not really. Defaults are somewhat identical. And excluding test files is actually wrong... as we need them transpiled for Jest.

File: package.json (hunk 1 above)
Comment (swernerx) [reply package.json]: Done

File: package.json (hunk 3 above)
Comment (swernerx) [reply package.json]: Pinning happens via package-lock. No need for non-caret versions here. It was also mixed. Some were "pinned", others not.

File: package.json (hunk 4 above)
Comment (swernerx) [reply package.json:31]: That's still needed. Moved it into a config file instead. (But forgot to push it - was listed in gitignore).

File: tsconfig.json (hunk 2 above)
Comment (bastianeicher) [reply tsconfig.json]: With `ts-jest` the test files get transpiled when running the tests. I excluded them from the regular build to avoid putting transpiled tests into the package published on NPM.

File: package.json (hunk 3 above)
Comment (bastianeicher) [reply package.json]: Regarding the mixing: Currently all `devDependencies` are pinned in `package.json` and none of the regular `dependencies` are pinned.

Regarding pinning the `devDependencies` (in addition to `package-lock.json`): I was following [the reasoning from the Renovate docs](https://docs.renovatebot.com/dependency-pinning/).

**Comment by swernerx** [comment]
I pushed a few changes. Hopefully it's better matching your requirements now.

**Comment by bastianeicher** [comment]
Thanks again. :)

Merged now, will publish a new release soon.
